core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/machineLearning/sklearnAdvanced/base/SklearnAdvancedBaseDesc.scala

```diff
@@ -0,0 +1,154 @@
+
+trait ParamClass {
+  def getName: String
+  def getType: String
+}
+
+abstract class SklearnMLOperatorDescriptor[T <: ParamClass] extends PythonOperatorDescriptor {
+  @JsonIgnore
+  def getImportStatements: String
+
+  @JsonIgnore
+  def getOperatorInfo: String
+
+  @JsonProperty(required = true)
+  @JsonSchemaTitle("Parameter Setting")
+  var paraList: List[HyperParameters[T]] = List()
+
+  @JsonProperty(required = true)
+  @JsonSchemaTitle("Ground Truth Attribute Column")
+  @JsonPropertyDescription("Ground truth attribute column")
+  @AutofillAttributeName
+  var groundTruthAttribute: String = ""
+
+  @JsonProperty(value = "Selected Features", required = true)
+  @JsonSchemaTitle("Selected Features")
+  @JsonPropertyDescription("Features used to train the model")
+  @AutofillAttributeNameList
+  var selectedFeatures: List[String] = _
+
+  private def getLoopTimes(paraList: List[HyperParameters[T]]): String = {
+    for (ele <- paraList) {
+      if (ele.parametersSource) {
+        return s"""table[\"${ele.attribute}\"].values.shape[0]"""
+      }
+    }
+    "1"
+  }
+
+  def getParameter(paraList: List[HyperParameters[T]]): List[String] = {
+    var str1 = ""; var str2 = ""; var str3 = ""
+    for (ele <- paraList) {
+      if (ele.parametersSource) {
+        str1 = str1 + String.format("%s = {},", ele.parameter.getName)
+        str2 =
+          str2 + String.format("%s(table['%s'].values[i]),", ele.parameter.getType, ele.attribute)
+        str3 = str3 + String.format(
+          "%s = %s(table['%s'].values[i]),",
+          ele.parameter.getName,
+          ele.parameter.getType,
+          ele.attribute
+        )
+      } else {
+        str1 = str1 + String.format("%s = {},", ele.parameter.getName)
+        str2 = str2 + String.format("%s ('%s'),", ele.parameter.getType, ele.value)
+        str3 = str3 + String.format(
+          "%s = %s ('%s'),",
+          ele.parameter.getName,
+          ele.parameter.getType,
+          ele.value
+        )
+      }
+    }
+    List(String.format("\"%s\".format(%s)", str1, str2), str3)
+  }
+
+  override def generatePythonCode(): String = {
+    val listFeatures = selectedFeatures.map(feature => s""""$feature"""").mkString(",")
+    val trainingName = getImportStatements.split(" ").last
+    val stringList = getParameter(paraList)
+    val trainingParam = stringList(1)
+    val paramString = stringList(0)
+    val finalCode =
+      s"""
+         |from pytexera import *
+         |
+         |import pandas as pd
+         |${getImportStatements}
+         |
+         |class ProcessTableOperator(UDFTableOperator):
+         |
+         |  @overrides
+         |  def process_table(self, table: Table, port: int) -> Iterator[Optional[TableLike]]:
+         |    global dataset
+         |    model_list = []
+         |    para_list = []
+         |    features = [$listFeatures]
+         |
+         |    if port == 0:
+         |      dataset = table
```

Comment:
what's the purpose of renaming a variable `a= b`?

Reply:
We need to get the dataset from port 0, the get the parameters from port 1.